### PR TITLE
WIP: Rnd 2

### PR DESCRIPTION
**Pull request by asdf2154s**
Random Network Distillation

### CODE REVIEW

File: python/ray/rllib/utils/rnd.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+        self._is_training_ph = is_training_ph
+
+        convfeat = 32
+        enlargement = 2
+        rep_size = 512
+                                            # 42 42 1
+        self.ob_rms = RunningMeanStd(shape=list(obs_space.shape[:2]) + [1], use_mpi=False)
+
+        # self.ph_mean = tf.placeholder(dtype=tf.float32, shape=(obs_ph.shape[1], obs_ph.shape[2], 1), name="obmean")
+        # self.ph_std = tf.placeholder(dtype=tf.float32, shape=(obs_ph.shape[1], obs_ph.shape[2], 1), name="obstd")
```
Comment (asdf2154s) [inline python/ray/rllib/utils/rnd.py:158]: Those variables should be used for input normalization

File: python/ray/rllib/utils/rnd.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+        # build target and predictor networks
+        logger.info("Building RND networks...")
+        with tf.variable_scope("rnd"):
+            with tf.variable_scope("target"):
+
+                xr = obs_ph
+                xr = tf.cast(xr, tf.float32)
+                xr = xr[:, :, :, -1:]
+                # xr = tf.clip_by_value((xr - self.ph_mean) / self.ph_std, -5.0, 5.0)
```
Comment (asdf2154s) [inline python/ray/rllib/utils/rnd.py:168]: Input normalization

File: python/ray/rllib/utils/rnd.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+                self._targets = fc(rgbr[0], 'fc1r', nh=rep_size, init_scale=np.sqrt(2))
+
+
+            self._targets = tf.stop_gradient(self._targets) # freeze target network
+
+            with tf.variable_scope("predictor"):
+                xr = obs_ph
+                xr = tf.cast(xr, tf.float32)
+                xr = xr[:, :, :, -1:]
+                # xr = tf.clip_by_value((xr - self.ph_mean) / self.ph_std, -5.0, 5.0)
```
Comment (asdf2154s) [inline python/ray/rllib/utils/rnd.py:184]: Input normalization

File: python/ray/rllib/utils/rnd.py [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+
+    def compute_intr_rew(self, obs):
+        self.ob_rms.update(obs[:, :, :, -1:])
+
+
+
+        feed_dict = {
+            self._obs_ph: obs,
+            # self.ph_mean: self.ob_rms.mean,
+            # self.ph_std: self.ob_rms.var ** 0.5,
```
Comment (asdf2154s) [inline python/ray/rllib/utils/rnd.py:224]: we need to find new way, how to feed those variables

**Comment by nikola-j** [comment]
Current version will only work with envs that are already normalised
